### DISCUSSION TITLE
Ability to add unit of measure?

### DISCUSSION THREAD

**Discussion by MaxVonEvil** [arendst/Tasmota#20990] [Q&A]
Hi all, not sure if this falls into the Tasmota feature request bucket, or it's more an HA thing, but here goes:

I've made a load-cell/weight with an HX711 amplifier for my water softener. Using Tasmota to integrate the smartified appliance into HomeAssistant.

Everything works fine, and two sensors created AbsRaw and WeightRaw with the Tasmota integration

![image](https://github.com/arendst/Tasmota/assets/19863636/11187b99-6211-49f0-a077-9f864fef47ef)

I'd like to use WeightRaw for my own nefarious purposes, however as neither of the two sensors report any unit of measure like the weight entity does, HA seems to interpret them as strings, i.e. it only displays a history bar, not recognizing it's a numeric value which can be graphed.

![image](https://github.com/arendst/Tasmota/assets/19863636/be322de9-0c12-4693-b3c7-931271da2c0f)

Now, I know I could just exclude this entity from the recorder and make a template sensor derived from the WeightRaw entity, however I wanted to ask the community if there's a better way to do this?

**Comment by sfromis** [comment]
Tasmota itself only provides units of measure for the very few values where you can customize what unit is being used, like temperature, speed or atmospheric pressure. HA defaults for all other units are supplied by the [Tasmota Integration](https://github.com/emontnemery/hatasmota/) based on the name of the value.

**Reply by sfromis** [reply]
Related thread: https://github.com/arendst/Tasmota/discussions/19666

**Comment by MaxVonEvil** [comment]
Thank you, initially trying to find out which tree to bark up against, be it Tasmota, the integration or HA :) - it seems to me that HA is lacking a blank generic UOM which could be manually mapped in the integrations. Anyway'll stop by the integration page and leave a comment as well. Thank you!
